lib/data/multipleBackendGateway.js

```diff
@@ -0,0 +1,68 @@
+import { errors } from 'arsenal';
+import { Logger } from 'werelogs';
+
+import config from '../Config';
+import parseLC from './locationConstraintParser';
+
+const logger = new Logger('MultipleBackendGateway', {
+    logLevel: config.log.logLevel,
+    dumpLevel: config.log.dumpLevel,
+});
+
+function createLogger(reqUids) {
+    return reqUids ?
+        logger.newRequestLoggerFromSerializedUids(reqUids) :
+        logger.newRequestLogger();
+}
+
+const clients = parseLC(config);
+
+const multipleBackendGateway = {
+    put: (stream, size, keyContext, backendInfo, reqUids, callback) => {
+        const controllingLocationConstraint =
+            backendInfo.getControllingLocationConstraint();
+        const client = clients[controllingLocationConstraint];
+        if (!client) {
+            const log = createLogger(reqUids);
+            log.error('no data backend matching controlling locationConstraint',
+            { controllingLocationConstraint });
+            return process.nextTick(() => {
+                callback(errors.InternalError);
+            });
+        }
+        return client.put(stream, size, keyContext,
+            reqUids, (err, key) => {
+                if (err) {
+                    const log = createLogger(reqUids);
+                    log.error('error from datastore',
+                             { error: err, implName: client.clientType });
+                    return callback(errors.InternalError);
+                }
+                const dataRetrievalInfo = {
+                    key,
+                    dataStoreName: controllingLocationConstraint,
+                };
+                return callback(null, dataRetrievalInfo);
+            });
+    },
+
+    get: (objectGetInfo, range, reqUids, callback) => {
+        const client = clients[objectGetInfo.dataStoreName];
+        if (client.clientType === 'scality') {
```

Comment:
I wonder why is the "scality" backend a special case, the adaptor code `const key = objectGetInfo.key ? objectGetInfo.key : objectGetInfo;` added in file and memory backends may be inserted in "scality" sproxydclient backend too right, or is it because we need backward compat with that other repo? I think sproxydclient is bundled with the release so I'm not sure we need such retro-compat if that's the idea here.